code/modules/mapping/mapping_helpers.dm

```diff
@@ -165,6 +165,223 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/effect/mapping_helpers)
 	else
 		airlock.abandoned = TRUE
 
+/obj/effect/mapping_helpers/airlock/access
```

Comment:
Since you are using airlock mapping helpers, you might as well use its proc `/obj/effect/mapping_helpers/airlock/proc/payload(obj/machinery/door/airlock/payload)` that exists specifically to be overriden by child procs to add behaviour.

Reply:
I used to, but I could not control it's priority then so they might conflict with req_access_txt or area codes.

Reply:
I was incorrect in our DMs, access is completely independant of initialisation as its done lazilly in gen_access. You can absolutely move the applicatino to the airlock mapping helper and remove it from the airlock itself.